### PR TITLE
truncate posted github messages so they fit

### PR DESCRIPTION
**Pull request by austin3dickey**
This PR truncates the github message in `d3515ecea53d_backfill_all_commits.py` so it always fits in the database.

### CODE REVIEW

File: conbench/entities/commit.py
Comment (jonkeane) [inline conbench/entities/commit.py:406]: Did we truncate these elsewhere before the recent changes? I imagine we had to have, right?

**Review by jonkeane — APPROVED** [review]
Thanks for this, and thanks for the test too. One question (mostly curiosity on my part)

File: conbench/entities/commit.py
Comment (austin3dickey) [reply conbench/entities/commit.py:406]: No! Isn't that crazy? We've just never ingested a commit with a message whose first line is over 250 characters. The backfill exposed one.

**Review by ElenaHenderson — APPROVED** [review]

File: conbench/entities/commit.py
Comment (jonkeane) [reply conbench/entities/commit.py:406]: Huh! I wonder if maybe the previous endpoint we were using automatically trimmed them and we didn't realize? 

I find it really surprising that we would only just now bump into it, but it is possible!

File: conbench/entities/commit.py
Comment (austin3dickey) [inline conbench/entities/commit.py:234]: This fixes another issue we were seeing. I thought a `s.exc.IntegrityError` would be raised, but our server was raising a `sqlalchemy.exc.PendingRollbackError`. So this try/except was not catching this expected error. The PendingRollbackError still has `"commit_index" in str(e)` so this change shouldn't cause any false negatives or false positives.

This is pretty hard to test, though.

**Review by jonkeane — APPROVED** [review]

File: conbench/entities/commit.py
Comment (austin3dickey) [reply conbench/entities/commit.py:406]: I just queried all commits to apache/arrow, and here's the distribution of length of the first line of commit messages. Only 4 commits ever were over 250 characters!

image

File: conbench/entities/commit.py
Comment (jgehrcke) [reply conbench/entities/commit.py:234]: > This fixes another issue we were seeing.

Would be great to have a super lightweight ticket about that one and to add a code comment in this over-generalized exception handler about _why_ it's over-generalized.

I think that it's also important to pull `backfilled_commits.append(commit)` out of this exception handler (it's just a regular list append, right?) just to make this code more readable.

Even if `if "commit_index" in str(e):` seems to be more or less precise it's always super valuable to log some exception details for future debugging/reasoning, and to be rather neutral when re-writing error messages (there is probably more than one type error message emitted by the stack that might after all contain the sub string `commit_index`)

Example, just to demonstrate briefly:

```python
try:
    commit = Commit.create_github_context(**commit_info)
except Exception as e:
    # Over-generalized exception handler so that this becomes a best effort approach.
    # GitHub commit could not be inserted into database, probably as of unique
    # key constraint. Regardless of the reason, proceed with the next commit.
    # Log the exact reason, however:
    log.info(f"commit {commit_info['sha']} could not be inserted as of error: ...")
    # next loop iteration (process next to-be-backfilled commit)
    continue

log.info(f"inserted commit {commit_info['sha']} to database")
backfilled_commits.append(commit)
```

Another related thought is that we maybe want to move the conflict error handling into the database, and keep interaction with the database at a minimum: one could try to insert all these commits in the _same_ database transaction, ignoring conflicts. Postgres has a feature for that, exposed via SQLAlchemy. I created a tiny issue for tracking that idea: https://github.com/conbench/conbench/issues/441.

Comment (austin3dickey) [reply conbench/entities/commit.py:234]: Agreed; I should have done this before I merged! I like #441 better, so I'll hold off on a followup PR until we tackle that issue.

**Pull request merged**
